`parseWithVerbForCurrentProcess` does not work in self-hosted

When using `parseWithVerbForCurrentProcess` we hit a compiler TODO:
```
$ zig build install -fno-stage1
/home/lee/src/zriscv/external/zig-args/args.zig:85:1: error: TODO (LLVM): implement const of pointer type '*(struct decl=Module.Decl.Index(4211))' (value.Value.Tag.u1_type)
fn parseInternal(comptime Generic: type, comptime MaybeVerb: ?type, args_iterator: anytype, allocator: std.mem.Allocator, error_handling: ErrorHandling) !ParseArgsResult(Generic, MaybeVerb) {
^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
error: zriscv...
error: The step exited with error code 1
error: the following build command failed with exit code 1:
/home/lee/src/zriscv/zig-cache/o/0c3f7ddebe856d7ecddec6a3fdc16368/build /home/lee/src/zig/build/zig /home/lee/src/zriscv /home/lee/src/zriscv/zig-cache /home/lee/.cache/zig install -fno-stage1 -fno-stage1
```

`parseForCurrentProcess` seems to work fine.

--- a/args.zig
+++ b/args.zig
@@ -136,19 +136,21 @@ fn parseInternal(comptime Generic: type, comptime MaybeVerb: ?type, args_iterato
                         const Tag = std.meta.Tag(Verb);
                         inline for (std.meta.fields(Verb)) |verb_info| {
                             if (verb.* == @field(Tag, verb_info.name)) {
-                                inline for (std.meta.fields(verb_info.field_type)) |fld| {
-                                    if (std.mem.eql(u8, pair.name, fld.name)) {
-                                        try parseOption(
-                                            verb_info.field_type,
-                                            result_arena_allocator,
-                                            &@field(verb.*, verb_info.name),
-                                            args_iterator,
-                                            error_handling,
-                                            &last_error,
-                                            fld.name,
-                                            pair.value,
-                                        );
-                                        found = true;
+                                if (comptime canHaveFieldsAndIsNotZeroSized(verb_info.field_type)) {
+                                    inline for (std.meta.fields(verb_info.field_type)) |fld| {
+                                        if (std.mem.eql(u8, pair.name, fld.name)) {
+                                            try parseOption(
+                                                verb_info.field_type,
+                                                result_arena_allocator,
+                                                &@field(verb.*, verb_info.name),
+                                                args_iterator,
+                                                error_handling,
+                                                &last_error,
+                                                fld.name,
+                                                pair.value,
+                                            );
+                                            found = true;
+                                        }
                                     }
                                 }
                             }
@@ -204,29 +206,31 @@ fn parseInternal(comptime Generic: type, comptime MaybeVerb: ?type, args_iterato
                                 const Tag = std.meta.Tag(Verb);
                                 inline for (std.meta.fields(Verb)) |verb_info| {
                                     const VerbType = verb_info.field_type;
-                                    if (verb.* == @field(Tag, verb_info.name)) {
-                                        const target_value = &@field(verb.*, verb_info.name);
-                                        if (@hasDecl(VerbType, "shorthands")) {
-                                            any_shorthands = true;
-                                            inline for (std.meta.fields(@TypeOf(VerbType.shorthands))) |fld| {
-                                                if (fld.name.len != 1)
-                                                    @compileError("All shorthand fields must be exactly one character long!");
-                                                if (fld.name[0] == char) {
-                                                    const real_name = @field(VerbType.shorthands, fld.name);
-                                                    const real_fld_type = @TypeOf(@field(target_value.*, real_name));
+                                    if (comptime canHaveFieldsAndIsNotZeroSized(VerbType)) {
+                                        if (verb.* == @field(Tag, verb_info.name)) {
+                                            const target_value = &@field(verb.*, verb_info.name);
+                                            if (@hasDecl(VerbType, "shorthands")) {
+                                                any_shorthands = true;
+                                                inline for (std.meta.fields(@TypeOf(VerbType.shorthands))) |fld| {
+                                                    if (fld.name.len != 1)
+                                                        @compileError("All shorthand fields must be exactly one character long!");
+                                                    if (fld.name[0] == char) {
+                                                        const real_name = @field(VerbType.shorthands, fld.name);
+                                                        const real_fld_type = @TypeOf(@field(target_value.*, real_name));
 
-                                                    // -2 because we stripped of the "-" at the beginning
-                                                    if (requiresArg(real_fld_type) and index != item.len - 2) {
-                                                        last_error = error.EncounteredUnexpectedArgument;
-                                                        try error_handling.process(error.EncounteredUnexpectedArgument, Error{
-                                                            .option = &option_name,
-                                                            .kind = .invalid_placement,
-                                                        });
-                                                    } else {
-                                                        try parseOption(VerbType, result_arena_allocator, target_value, args_iterator, error_handling, &last_error, real_name, null);
+                                                        // -2 because we stripped of the "-" at the beginning
+                                                        if (requiresArg(real_fld_type) and index != item.len - 2) {
+                                                            last_error = error.EncounteredUnexpectedArgument;
+                                                            try error_handling.process(error.EncounteredUnexpectedArgument, Error{
+                                                                .option = &option_name,
+                                                                .kind = .invalid_placement,
+                                                            });
+                                                        } else {
+                                                            try parseOption(VerbType, result_arena_allocator, target_value, args_iterator, error_handling, &last_error, real_name, null);
+                                                        }
+                                                        last_error = null; // we need to reset that error here, as it was set previously
+                                                        found = true;
                                                     }
-                                                    last_error = null; // we need to reset that error here, as it was set previously
-                                                    found = true;
                                                 }
                                             }
                                         }
@@ -286,6 +290,13 @@ fn parseInternal(comptime Generic: type, comptime MaybeVerb: ?type, args_iterato
 
     result.positionals = arglist.toOwnedSlice();
     return result;
+}
+
+fn canHaveFieldsAndIsNotZeroSized(comptime T: type) bool {
+    return switch (@typeInfo(T)) {
+        .Struct, .Union, .Enum, .ErrorSet => @sizeOf(T) != 0,
+        else => false,
+    };
 }
 
 /// The return type of the argument parser.
@@ -801,7 +812,7 @@ test "shorthand parsing (no verbs)" {
 
 test "basic parsing (with verbs)" {
     var titerator = TestIterator.init(&[_][:0]const u8{
-        "--output",  // non-verb options can come before or after verb
+        "--output", // non-verb options can come before or after verb
         "foobar",
         "booze", // verb
         "--with-offset",

--- a/build.zig
+++ b/build.zig
@@ -33,10 +33,20 @@ pub fn build(b: *std.build.Builder) void {
     run_demo_verb_2.addArgs(&[_][]const u8{
         "reload", "-f",
     });
+    const run_demo_verb_3 = demo_verb_exe.run();
+    run_demo_verb_3.addArgs(&[_][]const u8{
+        "forward",
+    });
+    const run_demo_verb_4 = demo_verb_exe.run();
+    run_demo_verb_4.addArgs(&[_][]const u8{
+        "zero-sized",
+    });
 
     const test_step = b.step("test", "Runs the test suite.");
     test_step.dependOn(&test_runner.step);
     test_step.dependOn(&run_demo.step);
     test_step.dependOn(&run_demo_verb_1.step);
     test_step.dependOn(&run_demo_verb_2.step);
+    test_step.dependOn(&run_demo_verb_3.step);
+    test_step.dependOn(&run_demo_verb_4.step);
 }

--- a/demo_verb.zig
+++ b/demo_verb.zig
@@ -36,6 +36,8 @@ pub fn main() !u8 {
                     .f = "force",
                 };
             },
+            forward: void,
+            @"zero-sized": struct {},
         },
         argsAllocator,
         .print,
@@ -69,6 +71,8 @@ pub fn main() !u8 {
                 });
             }
         },
+        .forward => std.debug.print("\t`forward` verb with no options received\n", .{}),
+        .@"zero-sized" => std.debug.print("\t`zero-sized` verb received\n", .{}),
     }
 
     std.debug.print("parsed positionals:\n", .{});

